src/java.base/share/classes/java/text/NumberFormat.java

```diff
@@ -58,103 +58,91 @@
 /**
  * {@code NumberFormat} is the abstract base class for all number
  * formats. This class provides the interface for formatting and parsing
- * numbers. {@code NumberFormat} also provides methods for determining
- * which locales have number formats, and what their names are.
+ * numbers in a {@link Locale localized} manner. This enables code that can be completely
+ * independent of the locale conventions for decimal points, thousands-separators,
+ * whether the number format is even decimal, or even the particular decimal
+ * digits used. For example, this class could be used within an application to
+ * produce a number in a currency format according to conventions of the locale
+ * a user resides in.
```

Comment:
Fair enough, I don't want to give any misconceptions.

The point was to give a general use case example of an app allowing format(s) tailored to the locale of the user. As I imagine this would be the most common case, rather than getting a locale separate than that of the user. I'll adjust the example.